src/check_jsonschema/cli/main_command.py

```diff
@@ -112,6 +113,11 @@ def pretty_helptext_list(values: list[str] | tuple[str, ...]) -> str:
     type=click.Choice(BUILTIN_SCHEMA_CHOICES, case_sensitive=False),
     metavar="BUILTIN_SCHEMA_NAME",
 )
+@click.option(
+    "--modeline-schema",
+    is_flag=True,
+    help="Use the schema defined in the modeline.",
+)
```

Comment:
Maybe `--yaml-modeline-schema`? Since this is very specific to YAML for now... or maybe going with this and raising a `NotImplementedError` on other content than yaml?